src/pages/profile/index.js

```diff
@@ -1,7 +1,264 @@
+import { useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import Avatar from '../../components/avatar';
+import { get, patch } from '../../service/apiClient';
+import useAuth from '../../hooks/useAuth';
+import jwtDecode from 'jwt-decode';
+import Form from '../../components/form';
+import TextInput from '../../components/form/textInput';
+
+import './styles.css';
+
 const ProfilePage = () => {
+  const { id } = useParams();
+  const [profileUser, setProfileUser] = useState(null);
+  const [loggedInUser, setLoggedInUser] = useState(null);
+  const { token } = useAuth();
+  const [canEdit, setCanEdit] = useState(false);
+  const [formData, setFormData] = useState(null);
+  const [prevProfileUserId, setPrevProfileUserId] = useState(null); // New state variable
+
+  useEffect(() => {
+    const fetchUserData = async (profileId, setter) => {
+      try {
+        const response = await get(`users/${profileId}`);
+        setter(response.data);
+      } catch (error) {
+        console.error(`Error fetching user data for ID ${profileId}:`, error);
+      }
+    };
+
+    if (token) {
+      const loggedInId = jwtDecode(token).userId;
+      fetchUserData(id, setProfileUser); // Fetch profile user data
+      fetchUserData(loggedInId, setLoggedInUser); // Fetch logged-in user data
+    }
+  }, [id, token]);
+
+  useEffect(() => {
+    if (profileUser && loggedInUser) {
+      setCanEdit(profileUser.id === loggedInUser.id || loggedInUser.role === 'teacher');
+    }
+  }, [profileUser, loggedInUser]);
+
+  useEffect(() => {
+    if (profileUser) {
+      if (profileUser.id !== prevProfileUserId) {
+        // Initialize formData with profileUser data only if the user ID has changed
+        setFormData({ ...profileUser });
+        setPrevProfileUserId(profileUser.id);
+      }
+    }
+  }, [profileUser, prevProfileUserId]);
+
+  const handleInputChange = (e) => {
+    const { name, value } = e.target;
+    setFormData((prevData) => ({
+      ...prevData,
+      [name]: value
+    }));
+  };
+
+  const handleSave = async () => {
+    const changeData = {};
+    const excludedFields = ['id', 'password']; // Fields to exclude
+
+    Object.keys(formData).forEach((key) => {
+      if (!excludedFields.includes(key) && formData[key] !== profileUser[key]) {
+        changeData[key] = formData[key];
+      }
+    });
+
+    if (Object.keys(changeData).length > 0) {
+      try {
+        const response = await patch(`users/${profileUser.id}`, changeData);
```

Comment:
why are you not using the pattern we have in the repo to send API requests of writing the fetch directly in the service/apiClient.js?

https://github.com/boolean-uk/team-dev-client-ex-2410-team1/blob/main/src/service/apiClient.js